File: helium/src/main/groovy/com/stanfy/helium/handler/codegen/objectivec/ObjcEntitiesOptions.java
package com.stanfy.helium.handler.codegen.objectivec;

import com.stanfy.helium.handler.codegen.GeneratorOptions;

import java.util.HashMap;
import java.util.Map;

/**
 * Options for a handler that generated Obj-C entities.
 */
public class ObjcEntitiesOptions extends GeneratorOptions {

  /** Class names prefix. */
  private String prefix = "HE";

  /** Map that contains mappings for custom Helium Types. i.e. timestamp -> NSDate.
   * It is used for generating custom(complex) types.
   * */
  private Map<String, String> customTypesMappings =  new HashMap<String, String>();

  public void setPrefix(final String prefix) {
    this.prefix = prefix;
  }

  /** Returns prefix for all files those would be generated */
  public String getPrefix() {
    return prefix;
  }


  public void setCustomTypesMappings(final Map<String, String> customTypesMappings) {
    this.customTypesMappings = customTypesMappings;
  }

  public Map<String, String> getCustomTypesMappings() {
    return customTypesMappings;
  }
}

File: codegen/objc/objc-entities/src/main/kotlin/com/stanfy/helium/handler/codegen/objectivec/entity/ObjCEntitiesOptions.kt
package com.stanfy.helium.handler.codegen.objectivec.entity;

import com.stanfy.helium.handler.codegen.GeneratorOptions


enum class ObjCMappingOption {
  NONE,
  MANTLE,
  SFMAPPING
}

/**
 * Options for a handler that generated Obj-C entities.
 */
class ObjCEntitiesOptions : GeneratorOptions() {

  /** Class names prefix. */
  var prefix = "HE";

  /**
   * Map that contains mappings for custom Helium Types. i.e. timestamp -> NSDate.
   * It is used for generating custom(complex) types.
   */
  var customTypesMappings = mapOf<String, String>()

  /**
   * Map that contains custom value transformers for Mantle entities generators
   * This is useful, when JSON have some strange format, and we cannot be sure about it
   * structure, so custom transformers are needed for class
   * This is only for mantle mappings
   */
  var mantleCustomValueTransformers = mapOf<String, String>()

  /**
   * Setting that allows to choose which mappings to generate
   * Default value is none, some general mappers can be used
   */
  var mappingsType : ObjCMappingOption = ObjCMappingOption.NONE

}
